### PR TITLE
Extract paged data from list responses

### LINKED ISSUE #118: API Pagination Breaks Lists
With pagination now [supported server-side](https://github.com/redhat-ipaas/ipaas-api-java/pull/58), the lists on the client are now broken.

### PR DESCRIPTION
**Pull request by jimmidyson**
Fixes #118

### CODE REVIEW

**Comment by kahboom** [comment]
@jimmidyson - A check failed, do you still want me to merge it in? Per Jenkins it looks like we need to set a CHROME_BIN env var for Karma to run: `No binary for Chrome browser on your platform.
  Please, set "CHROME_BIN" env variable.`

Also, typical errors with `fsevents` since we are running on Linux, and `fsevents` should only be run on OS X or macOS, but npm can be stupid like that:

```
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/angular-cli/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.17: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
```

I think there is a way to circumvent that, but not worth it considering it's just a warning and not the actual cause of the build failure.

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by kahboom** [comment]
@jimmidyson - What happened to being terse? ;D

File: src/app/app.module.ts [hunk 1]
```diff
@@ -15,6 +15,18 @@ import { ConfigService, configServiceInitializer } from './config.service';
 
 export function restangularProviderConfigurer(restangularProvider: any, config: ConfigService) {
   restangularProvider.setBaseUrl(config.getSettings().apiEndpoint);
+  restangularProvider.addResponseInterceptor((data: any, operation: string) => {
+    if (operation === 'getList' && Array.isArray(data.items)) {
```
Comment (kahboom) [inline src/app/app.module.ts]: Thank you for the `===`.

File: src/app/app.module.ts [hunk 2]
```diff
@@ -15,6 +15,18 @@ import { ConfigService, configServiceInitializer } from './config.service';
...
+      const pagingData = data.items;
+      if (!!pagingData.totalCount) {
```
Comment (kahboom) [inline src/app/app.module.ts]: And the `!!` as well.

**Comment by kahboom** [comment]
Thanks @jimmidyson 👍

**Pull request merged**